qp_klp/Amplicon.py

```diff
@@ -81,23 +86,59 @@ def quality_control(self):
                 new_path = join(output_folder, basename(raw_fastq_file))
                 copyfile(raw_fastq_file, new_path)
 
-    def generate_reports(self, input_file_path):
-        super()._generate_reports(self.pipeline.mapping_file)
+    def generate_reports(self):
+        super()._generate_reports()
         return None  # amplicon doesn't need project names
 
+    def _get_data_type(self, prep_file_path):
+        metadata = Step.parse_prep_file(prep_file_path)
+        if 'target_gene' in metadata.columns:
+            tg = set(metadata['sample target_gene'])
+            if len(tg) != 1:
+                raise ValueError("More than one value for target_gene")
+            else:
+                tg = tg.pop()
+
+            data_type = None
+            for key in Step.AMPLICON_SUB_TYPES:
+                # the key sub_types are often substrings of tg,
+                # rather than equal to tg.
+                if key in tg:
+                    data_type = key
+
+            if data_type is None:
+                raise ValueError(f"Valid data-types "
+                                 f"{Step.AMPLICON_SUB_TYPES} could not"
+                                 " be found within target_gene value "
+                                 f"'{tg}'")
+
+            return data_type
+        else:
+            raise ValueError("'target_gene' column not present in "
+                             "generated prep-files")
+
+    def generate_touched_studies(self, qclient):
+        results = {}
+        for study_id in self.prep_file_paths:
+            for prep_file_path in self.prep_file_paths[study_id]:
```

Comment:
This is a dictionary? It looks like the iter is doing that, so why not use items()?

Reply:
As far as I understand it, I could call self.prep_file_paths.items() and get back a (k,v) pair, but the v in this case will still be a list of prep-file-paths that I need to loop through. I'm not sure what the gain is.

Reply:
That you don't need to index the dictionary twice! Once to get the keys and another to retrieve the value of the key ...